Sherlock/pubsub.h

```diff
@@ -114,11 +118,21 @@ SOFTWARE.
 namespace current {
 namespace sherlock {
 
-template <typename E, JSONFormat J = JSONFormat::Current>
-class PubSubHTTPEndpointImpl {
+template <typename E, template <typename> class PERSISTENCE_LAYER, JSONFormat J>
+class PubSubHTTPEndpointImpl : public AbstractSubscriberObject {
  public:
-  explicit PubSubHTTPEndpointImpl(Request r)
-      : http_request_(std::move(r)), http_response_(http_request_.SendChunkedResponse()) {
+  using stream_data_t = StreamData<E, PERSISTENCE_LAYER>;
+
+  PubSubHTTPEndpointImpl(const std::string& subscription_id, ScopeOwned<stream_data_t>& data, Request r)
+      : data_(data, [this]() { time_to_terminate_ = true; }),
+        http_request_(std::move(r)),
+        http_response_(http_request_.SendChunkedResponse(
+            HTTPResponseCode.OK,
+            current::net::constants::kDefaultJSONContentType,
+            current::net::http::Headers({
+                {kSherlockHeaderCurrentSubscriptionId, subscription_id},
+                {kSherlockHeaderCurrentStreamSize, current::ToString(data_->persistence.Size())},
```

Comment:
Discussed to leave it as is (and add some `X-Current-Most-Recent-Timestamp` later).